### PR TITLE
Improves:  better experiment-end mechanism #161 

### PR DESCRIPTION
**Pull request by lckr**
Doesn't check for exp-end during shutdown

### CODE REVIEW

**Comment by lckr** [comment]
@LeanderK does the ExperimentPlatformState get set on endExperiment automatically, or do I need to change it manually?

**Comment by LeanderK** [comment]
@lucaskrauss answer your telegram messages

**Comment by marcelhollerbach** [comment]
hahaha

**Comment by lckr** [comment]
You can use the laugh-reaction now as well @marcelhollerbach .
Makes it even funnier!

**Comment by LeanderK** [comment]
@lucaskrauss push

**Comment by lckr** [comment]
Still need to determine the experiments to recover on boot and invoke the corresponding method

**Comment by LeanderK** [comment]
should be a piece of cake!

**Comment by LeanderK** [comment]
with strawberries!

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/ExperimentOperator.java [hunk 1]
```diff
@@ -62,27 +71,91 @@ public void startExperiment(Experiment experiment) {
...
+            }
+            experimentsPlatformOperations.setGlobalPlatformStatus(experiment,ExperimentsPlatformStatusPlatformStatus.stopped);
+            EventManager.EXPERIMENT_CHANGE.emit(new ChangeEvent<>(experiment,experimentFetcher.fetchExperiment(experiment.getId())));
+        }
+        /**
+        * Waits the remaining time of the shutdown process until stopping the experiment and setting its state accordingly
+        */
+         public void runRemaining(int minutesToShutdown){ //Used by a recover-method
+            this.minutesToShutdown = minutesToShutdown;
+            run();
```
Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/ExperimentOperator.java]: You sure this works ? if you call runRemaining blocking you are calling run blocking which will lock the calling thead for <2h

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/ExperimentOperator.java [hunk 2]
```diff
@@ -62,27 +71,91 @@ public void startExperiment(Experiment experiment) {
...
+                try {
+                    platformManager.unpublishTask(population.getPlatformId(), experiment).join();
+                } catch (PreActionException | CompletionException e) {
+                    log.fatal("Failed to unpublish experiment " + experiment + " on platform " + population.getPlatformId(), e);
+                }
+            }
+
+            experimentsPlatformOperations.setGlobalPlatformStatus(experiment, ExperimentsPlatformStatusPlatformStatus.shutdown);
+            ShutdownRunner runner = new ShutdownRunner(experiment);
+            runner.run();
```
Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/ExperimentOperator.java]: You are calling run. This changes absolutly nothing... its still blocking, call start.

**Comment by marcelhollerbach** [comment]
I am sorry.

**Comment by LeanderK** [comment]
@lucaskrauss whats the status?

**Comment by lckr** [comment]
Need to run it in docker once and test db-ops

**Pull request merged**